### PR TITLE
Elastic Search Docs - Add information about required permissions for Mattermost service account

### LINKED ISSUE #3992: Help Wanted: Elastic Search Docs - Add information about required permissions for Mattermost service account
Mattermost user `stu.doherty` from https://community-daily.mattermost.com has requested the following be documented:

```
This seems to work for basic indexing individual posts, full re-index, and purge (I don't lose the roles/users entries I had created):
```json
{
    "cluster": [
        "monitor" // ??????
    ],
    "indices": [
        {
            "names": [
                "channels",
                "users",
                "posts_*"
            ],
            "privileges": [
                "all"
            ],
            "allow_restricted_indices": false
        }
    ],
    "applications": [],
    "metadata": {
        "id": true,
        "index": true
    }
}
```

Now I think I need to ensure the following things work:
* Index roll-up
* ?

Not sure the easiest way to test this on local dev.  Would our unit tests help?
```

See the original post [here](https://community-daily.mattermost.com/_redirect/pl/861i6skr53njjbiptzg4gu5ygc).

_This issue was generated from [Mattermost](https://mattermost.com) using the [Doc Up](https://github.com/jwilander/mattermost-plugin-docup) plugin._

### PR DESCRIPTION
**Pull request by TomerPacific**
#### Summary
Following the discussion in the issue and also on the [Mattermost Hacktoberfest channel](https://community.mattermost.com/core/pl/poccsou7c7btpjhak5r68z58mo), it was decided to add another section to the [Elasticsearch section](https://docs.mattermost.com/scale/elasticsearch.html) of the documentation about the required permissions necessary for Mattermost.

This PR's aim is to address this by adding a description of the use case and relevant JSON examples of how those permissions look like.

#### Ticket Link
Fixes #3992

### CODE REVIEW

**Comment by sadohert** [comment]
@TomerPacific this is great... thanks so much!

I suggested 2 slight edits to make it generic to any customer (the string i replace are specific to a certain customer).

Feel free to adapt my edits to be more idiomatic for our docs when it comes to a variable subsitution.

@cwarnermm  - One thing to note, I don't see the `json` strings in the edit when I view on the preview environment.  Not sure if that's expected based on the markup used?

image

File: source/scale/elasticsearch.rst [hunk 1]
```diff
@@ -200,3 +200,56 @@ My Elasticsearch indexes won't complete, what should I do?
...
+          "indices:data/read*",
+          "indices:data/write*"
+        ]
+      }
+    ]
+  }
+
+A simpler, more flexible, and resilient variant of the above would be:
+
+.. code-block:: none
```
Comment (cwarnermm) [inline source/scale/elasticsearch.rst:237]:
```suggestion
.. code-block:: json
```

File: source/scale/elasticsearch.rst [hunk 2]
```diff
@@ -200,3 +200,56 @@ My Elasticsearch indexes won't complete, what should I do?
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 If you have an Elasticsearch indexing job that's paused, it's likely your Elasticsearch server has restarted. If you restart your Elasticsearch server, you must also restart Mattermost to ensure jobs are completed. If restarting the Mattermost server does not resolve the issue, please contact Mattermost support.
+
+Required Permissions For Mattermost Service Account
+---------------------------------------------------
+In "least privilege" environments you may need to further constrain the service account permissions to limit the access your Elasticsearch service account has. 
+The following JSON provides an example of a "least privilege" permission set that allows Mattermost to operate correctly with Elasticsearch:
+
+ .. code-block:: none
```
Comment (cwarnermm) [inline source/scale/elasticsearch.rst:209]:
```suggestion
 .. code-block:: json
```

**Comment by TomerPacific** [comment]
@sadohert - thanks for the feedback. I also saw the code blocks not being generated, so I made a revision, but that was after the previous commit was uploaded to a preview environment. Hopefully my changes fixed this.

**Comment by sadohert** [comment]
> @sadohert - thanks for the feedback. I also saw the code blocks not being generated, so I made a revision, but that was after the previous commit was uploaded to a preview environment. Hopefully my changes fixed this.

Thanks!  I'm still not seeing the json in the latest preview environment - http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/6732

Are you saying another commit is coming?

**Comment by cwarnermm** [comment]
I too am not seeing the code blocks and am currently investigating.

**Review by cwarnermm — APPROVED** [review]
Thank you for taking on a long-standing docs issue in our backlog, @TomerPacific!!!

**Comment by TomerPacific** [comment]
Thank you both @sadohert and @cwarnermm. I'll correct the indentation.

**Pull request merged**